README.md

```diff
@@ -1,2 +1,110 @@
+| endpoint                    | tcp://127.0.0.1:10000/                            | unix:///var/lib/csi/sockets/pluginproxy/csi.sock    | added to all volumes, for checking if a given volume was already created so that ControllerPublish/CreateVolume is idempotent. |
+| volume-attach-limit         | 1,2,3 ...                                         | -1                                                  | Value for the maximum number of volumes attachable per node. If specified, the limit applies to all nodes. If not specified, the value is approximated from the instance type.    |
+| debug           | true                                              | false                                               | if true, driver will enable the debug log level|
+
+
+# IBM PowerVS Block CSI Driver on Kubernetes
+Following sections are Kubernetes specific. If you are Kubernetes user, use followings for driver features, installation steps and examples.
+
+## Kubernetes Version Compatibility Matrix
+| PowerVS CSI Driver \ Kubernetes Version| v1.22 |
+|----------------------------------------|-------|
+| main branch                          | yes   |
+
+## Features
+* **Static Provisioning** - create a new or migrating existing PowerVS volumes, then create persistence volume (PV) from the PowerVS volume and consume the PV from container using persistence volume claim (PVC).
+* **Dynamic Provisioning** - uses persistence volume claim (PVC) to request the Kuberenetes to create the PowerVS volume on behalf of user and consumes the volume from inside container.
+* **Mount Option** - mount options could be specified in persistence volume (PV) to define how the volume should be mounted.
+* **[Volume Resizing](https://kubernetes-csi.github.io/docs/volume-expansion.html)** - expand the volume size. The corresponding CSI feature (`ExpandCSIVolumes`) is beta since Kubernetes 1.16.
+
+## Prerequisites
+* If you are managing PowerVS volumes using static provisioning, get yourself familiar with [Power Virtual Servers](https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-getting-started).
+* Get yourself familiar with how to setup Kubernetes on IBM Cloud and have a working Kubernetes cluster:
+  * Enable flag `--allow-privileged=true` for `kubelet` and `kube-apiserver`
+  * Enable `kube-apiserver` feature gates `--feature-gates=CSINodeInfo=true,CSIDriverRegistry=true,CSIBlockVolume=true`
+  * Enable `kubelet` feature gates `--feature-gates=CSINodeInfo=true,CSIDriverRegistry=true,CSIBlockVolume=true`
+
+
+## Installation
+#### Set up driver permission
+
+* Using secret object - Generate IBMCLOUD_APIKEY from the UI, put that user's credentials in [secret manifest](../deploy/kubernetes/secret.yaml), then deploy the secret
+```sh
+curl https://raw.githubusercontent.com/kubernetes-sigs/ibm-powervs-block-csi-driver/blob/main/deploy/kubernetes/secret.yaml > secret.yaml edit the IBMCLOUD_API_KEY
```

Comment:
`edit the IBMCLOUD_API_KEY` can be shifted to next line with `#` prefixed

Reply:
resolved